### ISSUE TITLE
Disable post download setting

### ISSUE THREAD

**Issue by tenebrios** [mastodon/mastodon#29478]
### Pitch

In our settings there should be a setting to disable/ turn off the ability for others to download posts be it photo, video or gif.

### Motivation

I believe everyone would benefit from this as it classifies as a privacy setting.

**Comment by tenebrios** [comment]
@flashymittens
Explain the 👎🏻

**Comment by tenebrios** [comment]
@finefindus 
Explain the 😄

**Comment by FineFindus** [comment]
You can already control who can see your posts through the visibility setting (i.e., setting it to followers only or mentioned people only).

A setting to prevent others from downloading is a band-aid solution at best, and useless at worst. Even if the web UI and protocol supported such a setting (which they don't), nothing could prevent a third-party application or another Fediverse server from re-implementing a download setting. From a technical perspective, the difference between displaying a user's media and downloading the media is only how long and where you store the data. So the setting would be impossible to actually enforce.
And even if it were somehow implemented and enforced, it would prevent someone from just screenshotting the media.

TLDR: If you don't want people to download things you've posted, don't post them publicly.

**Comment by tenebrios** [comment]
> You can already control who can see your posts through the visibility setting (i.e., setting it to followers only or mentioned people only).
> 
> A setting to prevent others from downloading is a band-aid solution at best, and useless at worst. Even if the web UI and protocol supported such a setting (which they don't), nothing could prevent a third-party application or another Fediverse server from re-implementing a download setting. From a technical perspective, the difference between displaying a user's media and downloading the media is only how long and where you store the data. So the setting would be impossible to actually enforce. And even if it were somehow implemented and enforced, it would prevent someone from just screenshotting the media.
> 
> TLDR: If you don't want people to download things you've posted, don't post them publicly.

You really rambled on there like a poindexter.
I'm not asking for anything to be "enforced".
I'm asking for a setting so the user can decide if they want it on or off.
No different than if the turn the gif auto play on or off setting.

Sure they could take a screenshot of a photo but this setting would disable download more specifically pertaining to video posts.
I don't think people will take screenshots of every  frame and then piece them together into a video format ;)

As for your "TLDR"... 🤦🏻‍♂️

**Comment by YJSoft** [comment]
You cannot "disable" download without putting it publicly.

If browser can display it, it can be downloaded. Any other ways to prevent download is all meaningless.

**Comment by YJSoft** [comment]
You cannot "disable" download without putting it publicly.

> If browser can display it, it can be downloaded. Any other ways to prevent download is all meaningless.(Blocking right click heavily influence UX and can bypassed easily by extentions)

**Comment by tenebrios** [comment]
> You cannot "disable" download without NOT putting it publicly.
> 
> If browser can display it, it can be downloaded. Any other ways to prevent download is all meaningless.

What are you talking about?
There are other apps with a disable download setting. No reason at all why that setting can't be here as well
